public/javascripts/SVLabel/src/SVLabel/canvas/ContextMenu.js

```diff
@@ -231,6 +231,16 @@ function ContextMenu (uiContextMenu) {
                 self.labelTags.forEach(function (tag) {
                     if (tag.tag === tagValue) {
                         if (!labelTags.includes(tag.tag_id)) {
+                            // Strings of alternate route present and no alternate route
+                            var alternateRoutePresentStr = 'alternate route present';
+                            var noAlternateRouteStr = 'no alternate route';
+                            // Automatically deselect one of the tags above if the other one is selected
+                            if (tagValue == alternateRoutePresentStr) {
+                                removeLabelAndUpdateUI(noAlternateRouteStr, labelTags);
```

Comment:
So now we have this way of removing labels and updating the UI, but in the else clause here we do a similar thing. Can you make it so that we use the same function in the else clause? You may have to modify your function to make it more generalizable.

Reply:
They're not doing exactly the same thing (tho I truly appreciate your thought and I think it is good practice). The first two lines of the else clause are the same as the single line in the if clause of the function, but the function iterates over all labels in the DOM, potentially applying that operation multiple times, while the else clause applies it only once, without the overhead of the loop. Also, the else clause does not change the tag's background color; that is done after the entire if block.